Mobile Navigation - Video Streaming 4.7 Updates (#129)
smartdevicelink/sdl_android_guides: docs/Mobile Navigation/Video Streaming/index.md

}
```

!!! Note

> If you are obfuscating the code in your app, make sure to exclude your `SdlRemoteDisplay` class. For more information on how to do that, you can check [Proguard Guidelines](/guides/android/proguard-guidelines/).

Reviewer: ```suggestion
If you are obfuscating the code in your app, make sure to exclude your class that extends `SdlRemoteDisplay`. For more information on how to do that, you can check [Proguard Guidelines](/guides/android/proguard-guidelines/).
```

> If you are obfuscating the code in your app, make sure to exclude you `SdlRemoteDisplay` class from being obfuscated. For more information on how to do that, you can check [Proguard Guidelines](/guides/android/proguard-guidelines/).

Reviewer 2: ```suggestion
If you are obfuscating the code in your app, make sure to exclude your `SdlRemoteDisplay` class. For more information on how to do that, you can check [Proguard Guidelines](/guides/android/proguard-guidelines/).
```

!!!

> ## Managing the Stream

Reviewer 2: maybe we should give an example of a `MyDisplay` class. We reference it a lot.

> To start streaming video, you can use `VideoStreamingManager`. The method that needs to be called is `startRemoteDisplayStream(Context context, final Class<? extends SdlRemoteDisplay> remoteDisplay, final VideoStreamingParameters parameters, final boolean encrypted)`.

Reviewer: ```suggestion
The `VideoStreamingManager` can be used to start streaming video after the `SdlManager` has successfully been started. This is performed by calling the method `startRemoteDisplayStream(Context context, final Class<? extends SdlRemoteDisplay> remoteDisplay, final VideoStreamingParameters parameters, final boolean encrypted)`.
```

```java
if (sdlManager.getVideoStreamManager() != null) {
    sdlManager.getVideoStreamManager().start(new CompletionListener() {
        @Override
        public void onComplete(boolean success) {
            if (success) {
                sdlManager.getVideoStreamManager().startRemoteDisplayStream(getApplicationContext(), MyDisplay.class, null, false);
            } else {
                Log.e(TAG, "Failed to start video streaming manager");
            }
        }
    });
}
```

### Ending the Stream

> When the `HMIStatus` is back to `HMI_NONE` it is time to stop the stream. This is accomplished through a method `stopStreaming()`.

Reviewer: Does the developer need to listen for HMI events in this case? Or does the VideoStreamingManager take care of the stream based on HMI levels?

Author: The current implementation of `VideoStreamingManager` doesn’t stop streaming automatically when the app is in HMI_NONE. So the developer has to do that.
